### ISSUE TITLE
Memory leak under Linux

### ISSUE THREAD

**Issue by fredvs** [rchastain/eschecs#30]
Hello Roland.

The memory leak comes from `CommandLine`

Without using CommandLine, no memory leak.
Very strange because all is freed in `finalization LExpr.Free;`

For testing I replaced code with followed and no more memory leak.

```
//  if CommandLine.HasOption('coloring', LBool) then
   FColoring := LBool;
// if CommandLine.HasOption('font', LStr) then
  LFont := LStr;
//  if CommandLine.HasOption('language', LInt) then
  LLang := TLanguage(LInt);
//  if CommandLine.HasOption('movetime', LInt) then
  FMoveTime := LInt;
//  if CommandLine.HasOption('scale', LInt) then
 LScale := LInt;
//  if CommandLine.HasOption('style', LInt) then
 FStyle := TBoardStyle(LInt);
// if CommandLine.HasOption('chess960', LBool) then
 FChess960 := LBool;
```

**Comment by rchastain** [comment]
OK Fred, thank you. So there would be a problem with _RegExpr_ unit?

I make some tests and come back to you.

Regards.

**Comment by rchastain** [comment]
You were right, Fred. No more memory leak, after I commented out the calls to CommandLine unit. Good!
I think I will remake the unit, using another regex library. I come back to you, as soon as it is done.

**Comment by fredvs** [comment]
Yes, good idea, you even do not need any other regex library, `ParamStr(x)` is your friend.

**Comment by fredvs** [comment]
For example:

```
x := 1;
While ParamStr(x) <> '' do 
begin
   if ParamStr(x) = 'this' then dothat;
   inc(x);
end;
```

**Comment by rchastain** [comment]
Thank you Fred for the example.

I remade the _CommandLine_ unit without using _RegExpr_, and the problem is still here. So I don't know what to think. I continue to investigate.

Good night!

**Comment by fredvs** [comment]
Maybe eschecs (like me) is not too fan of command-line parameter for a GUI app?
...
Ooops, ok, ok, I go to sleep.

Good night Roland!

**Comment by rchastain** [comment]
I expected that you would say something like that. :)

I will think of another solution. Good night Fred!

**Comment by rchastain** [comment]
Hello Fred!

Done. _Eschecs_ no longer uses the beautiful _CommandLine_ unit. Instead of sending command parameters, the _Options_ application overwrites _eschecs.ini_ before starting _Eschecs_.

By the way, I believe I had forgotten a file in my commit of yesterday. It is repaired, and it should be possible now to start _Eschecs_ from any directory. (I have to do the same modification in the _Options_ application.)

I promised to do something for the eaten pieces but, as it it a new feature, maybe we could do it in a future version, 5.1.0 for example?

So, we have solved the memory leak. :)

Best regards.

Roland
